### PR TITLE
fix: Remove update_infra from Expedia provider

### PR DESCRIPTION
**Pull request by zabarn**
# What this PR does / why we need it:

`full_scan_for_deletion` was introduced to `redis.py` to help with deleting "tables". 

This means we don't need to keep `update_infra()` in the Expedia provider anymore.

# Which issue(s) this PR fixes:

# Misc

### CODE REVIEW

File: sdk/python/feast/repo_config.py [hunk 1]
```diff
@@ -243,6 +243,8 @@ def __init__(self, **data: Any):
         self._online_store = None
         if provider == "expedia":
             self.online_config = data.get("online_store", "redis")
+            if self.online_config["type"] == "redis":
+                self.online_config["full_scan_for_deletion"] = True
```
Comment (zabarn) [inline sdk/python/feast/repo_config.py:247]: Do we want to do this if the Redis config [already defaults](https://github.com/ExpediaGroup/feast/blob/22bcbc49b4c76c1152e64da8f496026e34f21a23/sdk/python/feast/infra/online_stores/redis.py#L79-L80) `full_scan_for_deletion` to true?

[True is what triggers deletion.](https://github.com/ExpediaGroup/feast/blob/22bcbc49b4c76c1152e64da8f496026e34f21a23/sdk/python/feast/infra/online_stores/redis.py#L171-L173)

File: sdk/python/feast/infra/online_stores/redis.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ class RedisOnlineStoreConfig(FeastConfigBaseModel):
     key_ttl_seconds: Optional[int] = None
     """(Optional) redis key bin ttl (in seconds) for expiring entities"""
 
-    full_scan_for_deletion: Optional[bool] = True
+    full_scan_for_deletion: Optional[bool] = False
```
Comment (EXPEbdodla) [inline sdk/python/feast/infra/online_stores/redis.py:79]: Set this to Feast default i.e., `True`

File: sdk/python/feast/repo_config.py [hunk 3]
```diff
@@ -242,7 +242,17 @@ def __init__(self, **data: Any):
...
-            self.online_config = data.get("online_store", "redis")
+            self.online_config = data.get(
+                "online_store",
+                {
+                    "type": "redis",
+                    "connection_string": "${REDIS_CONNECTION_STRING}",
+                    "redis_type": "redis_cluster",
+                    "key_ttl_seconds": 604800,
+                },
+            )
```
Comment (EXPEbdodla) [inline sdk/python/feast/repo_config.py:253]:
```suggestion
            self.online_config = data.get("online_store", "redis")
```

Comment (EXPEbdodla) [reply sdk/python/feast/repo_config.py:253]: Is online_config is string, it uses the defaults set. Check online_store property method below.

File: sdk/python/feast/repo_config.py [hunk 4]
```diff
@@ -242,7 +242,17 @@ def __init__(self, **data: Any):
...
+            if self.online_config["type"] == "redis":
+                self.online_config["full_scan_for_deletion"] = True
```
Comment (EXPEbdodla) [inline sdk/python/feast/repo_config.py:255]: This should be False.

File: sdk/python/tests/unit/infra/scaffolding/test_repo_config.py [hunk 5]
```diff
@@ -358,7 +358,7 @@ def test_repo_config_init_expedia_provider():
     )
     assert c.registry_config == "registry.db"
     assert c.offline_config["type"] == "spark"
-    assert c.online_config == "redis"
+    assert c.online_config["type"] == "redis"
```
Comment (EXPEbdodla) [inline sdk/python/tests/unit/infra/scaffolding/test_repo_config.py:361]: revert this.

File: sdk/python/feast/feature_store.py [hunk 6]
```diff
@@ -1288,6 +1288,9 @@ def materialize_incremental(
                 f" from {Style.BRIGHT + Fore.GREEN}{start_date.replace(microsecond=0).astimezone()}{Style.RESET_ALL}"
                 f" to {Style.BRIGHT + Fore.GREEN}{end_date.replace(microsecond=0).astimezone()}{Style.RESET_ALL}:"
             )
+            logger.info(
+                f"{feature_view.name} from {start_date.astimezone()} to {end_date.astimezone()}:"
```
Comment (EXPEbdodla) [inline sdk/python/feast/feature_store.py:1292]:
```suggestion
                f"Materializing {feature_view.name} from {start_date.astimezone()} to {end_date.astimezone()}"
```

File: sdk/python/feast/feature_store.py [hunk 7]
```diff
@@ -1363,6 +1366,9 @@ def materialize(
         for feature_view in feature_views_to_materialize:
             provider = self._get_provider()
             print(f"{Style.BRIGHT + Fore.GREEN}{feature_view.name}{Style.RESET_ALL}:")
+            logger.info(
+                f"{feature_view.name} from {start_date.astimezone()} to {end_date.astimezone()}:"
```
Comment (EXPEbdodla) [inline sdk/python/feast/feature_store.py:1370]:
```suggestion
                f"Materializing {feature_view.name} from {start_date.astimezone()} to {end_date.astimezone()}"
```

**Review by EXPEbdodla — APPROVED** [review]

**Pull request merged**